### PR TITLE
Holodeck fixes

### LINKED ISSUE #15376: Holodeck safeties cannot be turned off
Something fucky, wasn't exactly clear. Ask shaps if you need more info.

### PR DESCRIPTION
**Pull request by AnturK**
Fixes #15376
Fixes Power off not working
Fixes spawned list duplication causing runtimes.

### CODE REVIEW

File: code/modules/holodeck/computer.dm [hunk 1]
```diff
@@ -161,14 +161,17 @@
...
 			return
-		if((area in program_cache) || (emagged && (area in emag_programs)))
+		if(area == offline_program || (area in program_cache) || (emagged && (area in emag_programs)))
 			load_program(area)
 	else if("safety" in href_list)
 		var/safe = text2num(href_list["safety"])
+		emagged = !safe
 		if(!program)
 			return
+		if(safe && (program in emag_programs))
```
Comment (neersighted) [inline code/modules/holodeck/computer.dm]: WAKE ME UP

Comment (Dorsisdwarf) [inline code/modules/holodeck/computer.dm]: nooooo

**Comment by octareenroon91** [comment]
what is that, a stealth nerf to hostile AI?

**Comment by Zilenan91** [comment]
non-hostile AIs too, prevents plasma flooding on eff station 9 rounds out of 10 and also prevents the crew from grabbing laser guns from the holdout room during other modes.

**Comment by octareenroon91** [comment]
I don't get that though.  I see that new line as making harder for silicons to cover up after enabling dangerous programs, but I'm not really feeling it.

https://github.com/tgstation/-tg-station/pull/15464#discussion-diff-52755212R171

**Comment by AnturK** [comment]
No idea what are you talking about ? This restores the emag functionality to AI's that was removed by mistake.

File: code/modules/holodeck/computer.dm (hunk 1 above)
Comment (octareenroon91) [inline code/modules/holodeck/computer.dm]: what is so exciting about this line?

Comment (tkdrg) [inline code/modules/holodeck/computer.dm]: theres nothing wrong with this you memesters
the holodeck should be stoppable

**Pull request merged**